### DISCUSSION TITLE
Language Server

### DISCUSSION THREAD

**Discussion by Namek** [mint-lang/mint#203] [Answered]
"Jump to Definition" feature in IDEs would look the language as more mature to the general audience.

What is the state of https://github.com/mint-lang/mint/tree/language-server/src/lsp ?

Can I help you with that?

**Comment by gdotdesign** [comment]
The `language-server` branch is a little behind `master` but I think it can be brought up to date without much problems.

The current state is:
- the basic protocol works, a client can connect to the server
- resolving a workspace based on a single source file works (this was a major milestone, resolving by searching for the closest `mint.json` file)
- `textDocument/completion` works in some cases
- `textDocument/hover` returns some type signatures in some cases
- started working on `textDocument/didChange` but could not get it to work yet
- the other features are not implemented

There was a discussion of it on Spectrum :arrow_right: https://spectrum.chat/mint-lang/development/language-server-support~227be091-fb65-40fb-ab60-952be791b28d

If you would like to help, I can bring the branch up to date just let me know.

**Comment by Namek** [comment]
Thanks for the amswer.

I'm a bit overloaded currently, I did not predict this, sorry. But I'd like to explore this at some point, yes. I'll let you know then.

**Comment by Namek** [comment]
Just found this discussion, linking:

https://spectrum.chat/mint-lang/development/language-server-support~227be091-fb65-40fb-ab60-952be791b28d

**Comment by gdotdesign** [comment]
I'm locking this because in favour of #270.
